ruby/neo4j/driver/internal/security/security_plan_impl.rb

```diff
@@ -3,10 +3,90 @@ module Security
       class << self
+        def for_all_certificates(requires_hostname_verification, revocation_strategy)
+          ssl_context = javax.net.ssl.SSLContext.get_instance('TLS')
+          ssl_context.init(javax.net.ssl.KeyManager[0].new, [TrustAllTrustManager.new].to_java(javax.net.ssl.KeyManager), nil)
+          new(true, ssl_context, requires_hostname_verification, revocation_strategy)
+        end
+
+        def for_custom_ca_signed_certificate(cert_file, requires_hostname_verification, revocation_strategy)
+          ssl_context = config_ssl_context(cert_file, revocation_strategy)
+          new(true, ssl_context, requires_hostname_verification, revocation_strategy)
+        end
+
+        def for_system_ca_signed_certificate(requires_hostname_verification, revocation_strategy)
+          ssl_context = config_ssl_context(nil, revocation_strategy)
+          new(true, ssl_context, requires_hostname_verification, revocation_strategy)
+        end
+
         def insecure
           new(false, nil, false, RevocationStrategy::NO_CHECKS)
         end
+
+        private
+
+        def config_ssl_context(custom_cert_file, revocation_strategy)
+          trusted_key_store = java.security.KeyStore.get_instance(java.security.KeyStore.get_default_type)
+          trusted_key_store.load(nil, nil)
+
+          if !custom_cert_file.nil?
+            org.neo4j.driver.internal.util.CertificateTool.loadX509_cert(custom_cert_file, trusted_key_store)
+          else
+            load_system_certificates(trusted_key_store)
+          end
+
+          pkix_builder_parameters = java.security.cert.PKIXBuilderParameters.new(trusted_key_store, java.security.cert.X509CertSelector.new)
+          pkix_builder_parameters.setRevocationEnabled(org.neo4j.driver.internal.RevocationStrategy.requiresRevocationChecking(revocation_strategy))
+
+          if org.neo4j.driver.internal.RevocationStrategy.requiresRevocationChecking(revocation_strategy)
+            System.set_property('jdk.tls.client.enableStatusRequestExtension', true)
+            if revocation_strategy == org.neo4j.driver.internal.RevocationStrategy.VERIFY_IF_PRESENT
+              Security.set_property('ocsp.enable', true)
+            end
+          end
+
+          ssl_context = javax.net.ssl.SSLContext.get_instance('TLS')
+
+          trust_manager_factory = javax.net.ssl.TrustManager.get_instance(javax.net.ssl.TrustManagerFactory.get_default_algorithm)
+          trust_manager_factory.init(javax.net.ssl.CertPathTrustManagerParameters.new(pkix_builder_parameters))
+          ssl_context.init(javax.net.ssl.KeyManager[0].new, trust_manager_factory.get_trust_managers, nil)
+          ssl_context
+        end
+
+        def load_system_certificates(trusted_key_store)
+          temp_factory = javax.net.ssl.TrustManagerFactory.get_instance(javax.net.ssl.TrustManagerFactory.get_default_algorithm)
+          temp_factory.init(java.security.KeyStore.java_class.cast(nil))
```

Comment:
```suggestion
          temp_factory.init(nil)
```
We assume all code will be in pure ruby, so we will have to replace all the java libraries. But at this point, we don't go that far.